From ac02a71a9cb8e80014218ba7de46f4f914b6e00c Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Tue, 30 May 2017 09:11:28 +0200
Subject: Revert "QQuickWindow::createTextureFromImage(): return nullptr for
 null images"

This reverts commit e6acf80136db9f667d0d4664f6c68065355d6811.

This breaks behavioral compatibility.

Task-number: QTBUG-61083
Change-Id: I0161d536502bab31aaf4ebc38f91e6c8842f72b0
Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
---
 src/quick/items/qquickwindow.cpp                   |  2 +-
 tests/auto/quick/qquickwindow/tst_qquickwindow.cpp | 11 -----------
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/src/quick/items/qquickwindow.cpp b/src/quick/items/qquickwindow.cpp
index ff4a357..c441cfc 100644
--- a/src/quick/items/qquickwindow.cpp
+++ b/src/quick/items/qquickwindow.cpp
@@ -3862,7 +3862,7 @@ QSGTexture *QQuickWindow::createTextureFromImage(const QImage &image) const
 QSGTexture *QQuickWindow::createTextureFromImage(const QImage &image, CreateTextureOptions options) const
 {
     Q_D(const QQuickWindow);
-    if (!isSceneGraphInitialized() || image.isNull()) // check both for d->context and d->context->isValid()
+    if (!isSceneGraphInitialized()) // check both for d->context and d->context->isValid()
          return 0;
     uint flags = 0;
     if (options & TextureCanUseAtlas)     flags |= QSGRenderContext::CreateTexture_Atlas;
diff --git a/tests/auto/quick/qquickwindow/tst_qquickwindow.cpp b/tests/auto/quick/qquickwindow/tst_qquickwindow.cpp
index 639027d..1d6547c 100644
--- a/tests/auto/quick/qquickwindow/tst_qquickwindow.cpp
+++ b/tests/auto/quick/qquickwindow/tst_qquickwindow.cpp
@@ -375,8 +375,6 @@ private slots:
 
     void testDragEventPropertyPropagation();
 
-    void createTextureFromImage();
-
 private:
     QTouchDevice *touchDevice;
     QTouchDevice *touchDeviceWithVelocity;
@@ -2833,15 +2831,6 @@ void tst_qquickwindow::testDragEventPropertyPropagation()
     }
 }
 
-void tst_qquickwindow::createTextureFromImage()
-{
-    // An invalid image should return a null pointer.
-    QQuickWindow window;
-    window.show();
-    QTest::qWaitForWindowExposed(&window);
-    QVERIFY(!window.createTextureFromImage(QImage()));
-}
-
 QTEST_MAIN(tst_qquickwindow)
 
 #include "tst_qquickwindow.moc"
-- 
cgit v1.0-4-g1e03

